tests/test_temporal.py

```diff
@@ -2356,3 +2377,754 @@ async def test_beta_graph_parallel_execution_in_workflow(client: Client):
+        plugins=[AgentPlugin(multi_model_selection_test_agent)],
+    ):
+        # Test using default model
+        output = await client.execute_workflow(
+            MultiModelWorkflowDefault.run,
+            args=['Hello'],
+            id='MultiModelWorkflowDefault',
+            task_queue=TASK_QUEUE,
+        )
+        assert output == 'Response from model 1'
+
+        # Test selecting second model
+        output = await client.execute_workflow(
+            MultiModelWorkflowSelectModel2.run,
+            args=['Hello'],
+            id='MultiModelWorkflowSelectModel2',
+            task_queue=TASK_QUEUE,
+        )
+        assert output == 'Response from model 2'
+
+        # Test selecting third model
+        output = await client.execute_workflow(
+            MultiModelWorkflowSelectModel3.run,
+            args=['Hello'],
+            id='MultiModelWorkflowSelectModel3',
+            task_queue=TASK_QUEUE,
+        )
+        assert output == 'Response from model 3'
+
+
+async def test_temporal_agent_multi_model_unregistered_error(allow_model_requests: None, client: Client):
+    """Test that using an unregistered model raises a helpful error."""
+    async with Worker(
+        client,
+        task_queue=TASK_QUEUE,
+        workflows=[MultiModelWorkflowUnregistered],
+        plugins=[AgentPlugin(multi_model_error_test_agent)],
+    ):
+        with workflow_raises(
+            UserError,
+            'Unregistered model instances cannot be selected at runtime inside a Temporal workflow. Register the model via `additional_models` or reference a registered model by name.',
+        ):
+            await client.execute_workflow(
+                MultiModelWorkflowUnregistered.run,
+                args=['Hello'],
+                id='MultiModelWorkflowUnregistered',
+                task_queue=TASK_QUEUE,
+            )
+
+
+async def test_temporal_agent_multi_model_string_selection_in_workflow(allow_model_requests: None, client: Client):
+    """Test that model selection works by passing different model instances."""
+    # Note: We can't test with model strings like 'test:gpt-4' because they're invalid.
+    # This test verifies that the model selection mechanism works correctly.
+    async with Worker(
+        client,
+        task_queue=TASK_QUEUE,
+        workflows=[MultiModelWorkflowSelectModel2String],
+        plugins=[AgentPlugin(multi_model_string_test_agent)],
+    ):
+        # Test selecting the second model
+        output = await client.execute_workflow(
+            MultiModelWorkflowSelectModel2String.run,
+            args=['Hello'],
+            id='MultiModelWorkflowSelectByInstance',
+            task_queue=TASK_QUEUE,
+        )
+        assert output == 'Second model response'
+
+
+async def test_temporal_agent_multi_model_backward_compatibility():
+    """Test that single-model agents work as before (backward compatibility)."""
```

Comment:
I don't think we need this test, as all the existing tests already verify the one-model case works.